src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,99 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final double DEFAULT_LOAD_FACTOR = 0.75;
+    private static final int CAPACITY_MULTIPLIER = 2;
+
+    private int capacity;
+    private int threshold;
+    private Node<K, V>[] storage;
+    private int size;
+
+    public MyHashMap() {
+        capacity = DEFAULT_CAPACITY;
+        threshold = (int) (capacity * DEFAULT_LOAD_FACTOR);
+        storage = new Node[capacity];
+    }
+
+    private static class Node<K, V> {
+        private final int hash;
+        private final K key;
+        private V value;
+        private Node<K, V> next;
+
+        public Node(K key, V value) {
+            hash = Math.abs(key == null ? 0 : key.hashCode());
+            this.key = key;
+            this.value = value;
+        }
+    }
 
     @Override
     public void put(K key, V value) {
-
+        Node<K, V> node = findNode(key);
+        if (node != null) {
+            node.value = value;
+        } else {
+            if (++size == threshold) {
```

Comment:
resize should be first in method put

Reply:
discussed in chat - decided to keep as it is